### ISSUE TITLE
Naming / syntax of top-level distribution modules / objects

### ISSUE THREAD

**Issue by bradcray** [chapel-lang/chapel#22904]
This issue is designed to check on some aspects of the naming and syntax choices around distributions as we look to Chapel 2.0.  For this discussion, I'll focus on the `Block` distribution as an example since it's the most stable and used, though similar naming choices apply to most of our standard distributions.

* The module is called `BlockDist`
  * this seems acceptable to leave as-is, though other obvious choices might be to:
  * call the module simply `Block`
  * or have it be a sub-module of some other module like `Dists.Block` (but how to do this without having to parse all standard distributions when any are used?)

* The top-level type is called `Block`
  * historically this was a class (wrapped by a `dmap`/`_distribution` record, but it is now a standalone record value on its own
  * is the capitalization OK?
    * on one hand, it's a record, and records are typically lowercase
    * on the other hand, it's a distributed object (it allocates a class per locale), and I'm a champion of using uppercase for distributed things (like domain/array variables) because:
      * in many respects being distributed is one of the most important qualities a variable can have in a Chapel program
      * it also tends to match up nicely with math where matrices are typically capital letters
      * we don't currently use capitals for any other variable types
      * so capitalizing distribution types/records could be viewed as being in-line with this philosophy
  * independent of capitalization, is the name OK or is there something better?
    
* Here are some ways that `Block` can be used with a `dmapped` clause today:
  * using a named distribution:
    ```chapel
    const Dist = new Block({1..10});
    const Dom = {1..10} dmapped Dist;
    ```
  * using an anonymous distribution:
    ```chapel
    const Dom = {1..10} dmapped new Block({1..10});
    ```
  * same, but without the `new` keyword:
    ```chapel
    const Dom = {1..10} dmapped Block({1..10});
    ```

  Of these three, the last seems convenient and reads pretty well, but it's also a bit odd (in the sense of "Why can `new` simply be omitted).  Are we OK with keeping it / all three forms?  [note that `dmapped` is currently unstable and has been discussed as being replaced with `distributed` or some other keyword, so this last bullet could be punted as being specific to that keyword].

**Comment by bradcray** [comment]
Personally, I'm OK with all of the current names/choices here though of all the items I list, the final one definitely feels the hardest to rationalize and is the one I'd be most likely to change (don't permit `dmapped Block` without a `new`).  I'm mostly opening this question to see if we have agreement and to not leave the questions unasked before it's too late.

For awhile I considered things like:
* Should we rename the `Block` type to `BlockDist` since that's what it really is ("a block distribution") but over time I haven't really liked this because I think things like `Block.createDomain(...)` and `dmapped [new] Block(...)` are nicer and more readable 
* Should we make `Block` into a standalone factory routine to preserve `dmapped Block(...)`?  But I don't like that this forces using something else for the type name and having to write `BlockNewName.createDomain(...)` rather than `Block.createDomain(...)`
* Should we rename the module to simply `Block`?  We can have a module and type with the same name, though I think it's less confusing when they don't.  And given the previous bullets, I wasn't really excited about renaming `Block`
* Should we rename `Block` to `block` since it's a record?  I'm open to this, but probably wouldn't do it if I were the only developer.

**Comment by bradcray** [comment]
Thinking on this another day, I'm having trouble deciding whether my preference for `Block` is a good one or whether I'm just too used to it being capitalized after years and years of use.  Specifically, even though I (strongly) prefer capitals for distributed domains and arrays, Block itself is a record (albeit a record whose implementation involves all locales, so it is distributed in that sense).

If we stuck with capital-B Block, a corollary to my rationale would be that layouts should be lowercase.

**Comment by bradcray** [comment]
Capturing a conversation with Michael w.r.t. the final bullet:

* He likes it because it reads nicer and would suggest defining it in the language spec as "`dmapped` can be used to invoke an initializer without explicitly typing the `new` keyword.

**Comment by lydia-duncan** [comment]
@bradcray - what's the next step here?

**Comment by bradcray** [comment]
I'm hoping to be far enough along to request a Monday meeting slot next week for item 2 above.  Based on that outcome, I'll decide whether to have a follow-up on item 1 or let sleeping dogs lie.  Item 3 is interesting, but since dmapped is unstable for 2.0, not necessarily an issue (unless we introduce a dmapped replacement, which I'd still like to do...).

**Comment by bradcray** [comment]
In a group meeting today, there was a majority preference for coming up with a new name (naming scheme) rather than `Block` or `block` for a distribution:

> independent of capitalization, is the name OK or is there something better?

I've forked this question off to https://github.com/chapel-lang/chapel/issues/23118

**Comment by lydia-duncan** [comment]
I have a vague memory of there being a decision here and us renaming the distributions in question.  Is that correct?  Is that work complete?

**Comment by bradcray** [comment]
I think you're right that this can be closed.  The first two points were addressed by keeping the module name and changing the distribution names to `blockDist`, say (the lowercase version of the module).  In the dmapped subteam this week, we talked about the final point here—not permitting `dmapped` to imply an automatic `new` keyword—but that should probably be split off into its own issue at this point.  Tagging @jeremiah-corrado to see if he could potentially file an issue for that if he wasn't already planning to.
